// bifrost-front-end-library/esm2015/lib/components/button/angular/button.component.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
import { Component, Input, Output, EventEmitter } from "@angular/core";
/**
 * API is the same between **Angular** and **Drupal**.
 *
 */
let ButtonComponent = class ButtonComponent {
    constructor() {
        /** Variant
          * @values csr
          */
        this.variant = '';
        /**
         * Handler to be called when the button is clicked by a user.
         *
         * Will also block the emission of the event if `disabled` is true.
         */
        this.click = new EventEmitter();
    }
    ngOnInit() {
        console.log('button', this);
    }
    onClick() {
        this.click.emit(true);
    }
};
__decorate([
    Input(),
    __metadata("design:type", Boolean)
], ButtonComponent.prototype, "exampleOfIgnore", void 0);
__decorate([
    Input(),
    __metadata("design:type", String)
], ButtonComponent.prototype, "label", void 0);
__decorate([
    Input(),
    __metadata("design:type", Boolean)
], ButtonComponent.prototype, "disabled", void 0);
__decorate([
    Input(),
    __metadata("design:type", String)
], ButtonComponent.prototype, "color", void 0);
__decorate([
    Input(),
    __metadata("design:type", String)
], ButtonComponent.prototype, "variant", void 0);
__decorate([
    Input(),
    __metadata("design:type", Boolean)
], ButtonComponent.prototype, "reversed", void 0);
__decorate([
    Input(),
    __metadata("design:type", Boolean)
], ButtonComponent.prototype, "fullWidth", void 0);
__decorate([
    Input(),
    __metadata("design:type", String)
], ButtonComponent.prototype, "icon", void 0);
__decorate([
    Input(),
    __metadata("design:type", String)
], ButtonComponent.prototype, "styleClass", void 0);
__decorate([
    Output(),
    __metadata("design:type", EventEmitter)
], ButtonComponent.prototype, "click", void 0);
ButtonComponent = __decorate([
    Component({
        selector: "bf-button",
        template: "<button \n    type=\"button\" \n    class=\"\n        btn btn-md \n        {{ color ? 'btn-' + color : '' }}\n        {{ fullWidth ? 'btn-block' : '' }}\n        {{ variant ? 'btn-' + variant : '' }}\n        {{ reversed ? 'btn-reversed' : '' }}\n        {{ styleClass }}\n    \"\n    [disabled]=\"disabled\"\n    (click)=\"onClick()\"\n>\n    {{ label }}\n    <ng-content></ng-content>\n</button>",
        styles: [".btn{min-height:40px;padding:0 1.5rem;color:#050504;font-family:BlenderPro,-apple-system,BlinkMacSystemFont,\"Segoe UI\",Roboto,\"Helvetica Neue\",Arial,sans-serif,\"Apple Color Emoji\",\"Segoe UI Emoji\",\"Segoe UI Symbol\";font-size:1rem;font-weight:500;letter-spacing:1px;text-transform:uppercase}.btn[disabled]{opacity:1}.btn-primary{border:none;background-color:#ffd000}.btn-primary:hover{color:currentColor;background-color:#dab201}.btn-primary-alt{color:#fff;background-color:#060605}.btn-primary-alt:hover{background-color:#363635}.btn-primary-alt.btn-reversed{color:#060605;background-color:#fff}.btn-primary-alt.btn-reversed:hover{color:#fff;background:#363635}.btn-csr[disabled],.btn-csr[disabled]:hover,.btn-primary-alt[disabled],.btn-primary-alt[disabled]:hover,.btn-primary[disabled],.btn-primary[disabled]:hover{color:#aaaaa3;background-color:#f8f8f7}.btn-csr[disabled].btn-reversed,.btn-csr[disabled]:hover.btn-reversed,.btn-primary-alt[disabled].btn-reversed,.btn-primary-alt[disabled]:hover.btn-reversed,.btn-primary[disabled].btn-reversed,.btn-primary[disabled]:hover.btn-reversed{color:#575754;background-color:rgba(47,47,47,.5)}.btn-secondary{color:#060605;border-color:#575754;background-color:transparent;will-change:box-shadow;transition:box-shadow .1s ease-in-out}.btn-secondary:not([disabled]):hover{color:currentColor;box-shadow:inset 0 0 0 1px #363635;border-color:#363635;background-color:transparent}.btn-secondary.btn-reversed{color:#fff;border-color:#fff}.btn-secondary.btn-reversed:not([disabled]):hover{color:#fff;border-color:#fff;box-shadow:inset 0 0 0 1px #fff}.btn-secondary.btn-reversed[disabled],.btn-secondary.btn-reversed[disabled]:hover{color:#575754;background-color:transparent;border-color:#575754}.btn-secondary[disabled],.btn-secondary[disabled]:hover{color:#aaaaa3;background-color:transparent;border-color:#e0e0da}.btn-tertiary{border:none;background-color:transparent}.btn-tertiary:hover{color:currentColor;background-color:#e6e6e6}.btn-tertiary.btn-reversed{color:#fff}.btn-tertiary.btn-reversed:hover{background-color:#2f2f2f}.btn-tertiary.btn-reversed[disabled],.btn-tertiary.btn-reversed[disabled]:hover{color:#575754;background-color:rgba(47,47,47,.5)}.btn-tertiary[disabled],.btn-tertiary[disabled]:hover{color:#aaaaa3;background-color:transparent}.btn-csr{color:#fff;border:none;background-color:#0b8339}.btn-csr:hover{background-color:#0b8339}"]
    }),
    __metadata("design:paramtypes", [])
], ButtonComponent);
export { ButtonComponent };
